lang/csharp/src/apache/main/IO/Parsing/Symbol.cs

```diff
@@ -0,0 +1,1049 @@
+            }
+
+            /// <summary>
+            /// Returns the symbol at the given index.
+            /// </summary>
+            public virtual Symbol GetSymbol(int index)
+            {
+                return Symbols[index];
+            }
+
+            /// <summary>
+            /// Returns the label at the given index.
+            /// </summary>
+            public virtual string GetLabel(int index)
+            {
+                return Labels[index];
+            }
+
+            /// <summary>
+            /// Returns the size.
+            /// </summary>
+            public virtual int Size()
+            {
+                return Symbols.Length;
+            }
+
+            /// <summary>
+            /// Returns the index of the given label.
+            /// </summary>
+            public virtual int FindLabel(string label)
+            {
+                if (!ReferenceEquals(label, null))
+                {
+                    for (int i = 0; i < Labels.Length; i++)
+                    {
+                        if (label.Equals(Labels[i]))
+                        {
+                            return i;
+                        }
+                    }
+                }
+
+                return -1;
+            }
+
+            /// <inheritdoc />
+            protected override Symbol Flatten(IDictionary<Sequence, Sequence> map,
+                IDictionary<Sequence, IList<Fixup>> map2)
+            {
+                Symbol[] ss = new Symbol[Symbols.Length];
+                for (int i = 0; i < ss.Length; i++)
+                {
+                    ss[i] = Symbols[i].Flatten(map, map2);
+                }
+
+                return new Alternative(ss, Labels);
+            }
+        }
+
+        /// <summary>
+        /// The error action.
+        /// </summary>
+        public class ErrorAction : ImplicitAction
+        {
+            /// <summary>
+            /// The error message.
+            /// </summary>
+            public readonly string Msg;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="Symbol.ErrorAction"/> class.
+            /// </summary>
+            public ErrorAction(string msg)
+            {
+                this.Msg = msg;
+            }
+        }
+
+        /// <summary>
+        /// Int check action.
+        /// </summary>
+        public class IntCheckAction : Symbol
+        {
+            /// <summary>
+            /// The size.
+            /// </summary>
+            public readonly int Size;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="Symbol.IntCheckAction"/> class.
+            /// </summary>
+            public IntCheckAction(int size) : base(Kind.ExplicitAction)
+            {
+                this.Size = size;
+            }
+        }
+
+        /// <summary>
+        /// The enum adjust action.
+        /// </summary>
+        public class EnumAdjustAction : IntCheckAction
```

Comment:
In the Java code, this class is used by `ResolvingGrammarGenerator`, which was the only class in the `parsing` package that was not ported, since it is not used by the JSON encoder/decoder.  Since `EnumAdjustAction` is not used, I will remove it.